### PR TITLE
Refactor SignatureValidator

### PR DESCRIPTION
**Pull request by charlesjhongc**
Only accept EOA with EIP-721 signature or Contract with EIP-1271 implementation. Make `SignatureValidator` as a library.

### CODE REVIEW

**Review by pilagod — APPROVED** [review]

File: test/SignatureValidator.t.sol [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    uint256 userPrivateKey = 1234;
+    uint256 walletAdminPrivateKey = 5678;
+    bytes32 digest = keccak256("EIP-712 data");
+    MockERC1271Wallet mockERC1271Wallet;
+
+    function setUp() public {
+        mockERC1271Wallet = new MockERC1271Wallet(vm.addr(walletAdminPrivateKey));
+    }
+
+    function testEIP712Signature() public {
```
Comment (NIC619) [inline test/SignatureValidator.t.sol:20]: Also test `_signerAddress` being `address(0)` and `_signature` being empty bytes?

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]
